pype/plugins/celaction/_unused_publish/submit_publish_job.py

```diff
@@ -0,0 +1,283 @@
+
+        - ext (str, Optional): The extension (including `.`) that is required
+            in the output filename to be picked up for image sequence
+            publishing.
+
+        - publishJobState (str, Optional): "Active" or "Suspended"
+            This defaults to "Suspended"
+
+    This requires a "startFrame" and "endFrame" to be present in instance.data
+    or in context.data.
+
+    """
+
+    label = "Submit image sequence jobs to Deadline"
+    order = pyblish.api.IntegratorOrder + 0.2
+
+    hosts = ["celaction"]
+
+    families = [
+        "render",
+        "deadline"
+    ]
+
+    def process(self, instance):
+
+        # DEADLINE_REST_URL = os.environ.get("DEADLINE_REST_URL",
+        #                                    "http://localhost:8082")
+        # assert DEADLINE_REST_URL, "Requires DEADLINE_REST_URL"
+
+        # Get a submission job
+
+        job = instance.data.get("deadlineSubmissionJob")
+        if not job:
+            raise RuntimeError("Can't continue without valid deadline "
+                               "submission prior to this plug-in.")
+        ################
+        ft_data = instance.context.data["ftrackData"]
+        project = ft_data['Project']['name']
+        project_code = ft_data['Project']['code']
+        projects_path = os.path.dirname(ft_data['Project']['root'])
+
+        data = instance.data.copy()
+        asset = instance.context.data["ftrackData"]['Shot']['name']
+        subset = 'render' + \
+            instance.context.data["ftrackData"]['Task']['name'].capitalize()
+
+        state = data.get("publishJobState", "Suspended")
+        # job_name = "{batch} - {subset} [publish image sequence]".format(
+        #     batch=job["Props"]["Name"],
+        #     subset=subset
+        # )
+        job_name = "{asset} [publish image sequence]".format(
+            asset=asset
+        )
+
+        # Get start/end frame from instance, if not available get from context
+        context = instance.context
+
+        start = int(instance.data['start'])
+        end = int(instance.data['end'])
+
+        try:
+            source = data['source']
+        except KeyError:
+            source = context.data["currentFile"]
+
+        # Write metadata for publish job
+        render_job = data.pop("deadlineSubmissionJob")
+        metadata = {
+            "asset": asset,
+            "regex": r"^.*\.png",
+            "subset": subset,
+            "startFrame": start,
+            "endFrame": end,
+            "fps": context.data.get("fps", None),
+            "families": ["render"],
+            "source": source,
+            "user": context.data["user"],
+            "version": context.data.get('version'),
+            "audio": context.data["audio"]['filename'],
+            # Optional metadata (for debugging)
+            "metadata": {
+                "instance": data,
+                "job": job,
+                "session": fake_avalon_session(project, projects_path)
```

Comment:
Fake session?

Reply:
I suppose this should be completely removed if it's unused.